### PR TITLE
half Vector /  half WMS layers

### LINKED ISSUE #2884: Create option for vector layers to display as WMS layers at low resolutions
**Is your feature request related to a problem? Please describe.**
Need a way for layers tor request WMS layers at low resolutions and vector tiles at high resolutions
**Describe the solution you'd like**
Create a field on the layer config called `maxRes` that is the maximum resolution that this layer will be shown as a vector.

**Additional context**
This capability is a performance related issue. When too many vector features are present, map interactions become slow.

### PR DESCRIPTION
**Pull request by Benjaki2**
## Description

Fixes #2884

Create breakpoint layer in layer config. 
- [x] Render higher zoom levels as vectors and lower zoom levels as wms layers to enhance performance
- [x] Simplify alerts (Bring current alerts into one container file and reuse functions)
- [x] Create UI that makes this feature clear (onclick Alerts, icons, etc)
- [x] add irrigation styling (@minniewong )
- [x] rebase on develop
- [x]  unit tests and e2e tests (working on it 😬)

## Further comments

There could have just been one value in the config called `resolutionBreakPoint`... I created the config options the way that I did so that it could potentially be implemented for other layer-types(hi-res wmts layers at higher zoom levels)in the future or you could add layer specific customizations for each layer.

Code Review => @jasontk19 
Functionality Testing => @minniewong 

## Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/master/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview

### CODE REVIEW

**Comment by Benjaki2** [comment]
Good find @minniewong ! Font-awesome must have made a minor release that changed that icon

File: web/js/map/util.js [hunk 1]
```diff
@@ -37,3 +39,37 @@ export function getActiveLayerGroup(map, layerGroupString) {
...
+    util.toISOStringSeconds(util.roundTimeOneMinute(date))
+  }&layer=${
+    layerName
+  }&tilematrixset=${
+    tileMatrixSet
+  }&Service=WMTS`
+  + '&Request=GetTile'
+  + '&Version=1.0.0'
+  + '&FORMAT=application%2Fvnd.mapbox-vector-tile'
+  + '&TileMatrix={z}&TileCol={x}&TileRow={y}';
```
Comment (jasontk19) [inline web/js/map/util.js:62]:
```suggestion
  const time = util.toISOStringSeconds(util.roundTimeOneMinute(date));
  const params = [
    `TIME=${time}`,
    `layer=${layerName}`,
    `tilematrixset=${tileMatrixSet}`,
    `Service=WMTS`,
    `Request=GetTile`,
    `Version=1.0.0`,
    `FORMAT=application%2Fvnd.mapbox-vector-tile`,
    `TileMatrix={z}`,
    `TileCol={x}`,
    `TileRow={y}`,
  ]
  return `?${params.join('&')}`;
```

Comment (jasontk19) [reply web/js/map/util.js:62]: This might be a little more readable.

File: web/js/modules/layers/util.js [hunk 2]
```diff
@@ -1225,3 +1225,66 @@ export function mapLocationToLayerState(
...
+ * vector layer
+ *
+ * @param {Array} activeLayers
+ *
+ * @return {Boolean}
+ */
+export const hasVectorLayers = (activeLayers) => {
+  const len = activeLayers.length;
+  // Used for loop
+  // so break could be used
```
Comment (jasontk19) [inline web/js/modules/layers/util.js:1239]: 👍  on this comment

File: web/js/components/feature-alert/vector-alert-modal.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { Container, Row, Col } from 'reactstrap';
+
+const VectorAlertModalBody = () => (
+
+  <Container className="event-alert-modal-body">
+    <Row className="p-2">
```
[the remark this answers is not in the document]
Comment (Benjaki2) [reply web/js/components/feature-alert/vector-alert-modal.js:7]: 👍  about the `h2` tags.

p-2 & p-3 are bootstrap padding classes

File: web/js/modules/alerts/reducer.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        isVectorAlertActive: false,
+      });
+    case ACTIVATE_VECTOR_ALERT:
+      return lodashAssign({}, state, {
+        isVectorAlertActive: true,
+      });
+    case REMOVE_LAYER:
+      if (!state.isVectorAlertActive) {
+        return state;
+      } if (!hasVectorLayers(action.layers)) {
```
[the remark this answers is not in the document]
Comment (Benjaki2) [reply web/js/modules/alerts/reducer.js:27]: This is something that `eslint` did... I guess that  `}if{` is short hand for else if {

**Comment by Benjaki2** [comment]
@jasontk19 thanks for the feedback and suggestions!

I ran `npm update` yesterday. I figured that would be a good time since we are about to go into testing. `npm update` respects semver. So, there shouldn't be any major updates, Just minors and patches. I could revert that though if you think it's a dangerous move. Let me know!

**Comment by jasontk19** [comment]
> @jasontk19 thanks for the feedback and suggestions!
> 
> I ran `npm update` yesterday. I figured that would be a good time since we are about to go into testing. `npm update` respects semver. So, there shouldn't be any major updates, Just minors and patches. I could revert that though if you think it's a dangerous move. Let me know!

Ah ok, figured it was something like that.  I wasn't familiar with `npm update` but that totally make sense and seems like a good time to do it!

File: web/js/components/feature-alert/vector-alert-modal.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jasontk19) [reply web/js/components/feature-alert/vector-alert-modal.js:7]: Not seeing any commits for this but, not sure if you meant to push an update for this?

[the remark this answers is not in the document]
Comment (Benjaki2) [reply web/js/components/feature-alert/vector-alert-modal.js:7]: I got denied by my push verify without realizing it (it's working :) ). It should be up now!

**Pull request merged**